src/cdk-experimental/testing/component-harness.ts

```diff
@@ -252,3 +258,78 @@ export interface ComponentHarnessConstructor<T extends ComponentHarness> {
    */
   hostSelector: string;
 }
+
+/** A class used to represent a ComponentHarness class with additional predicates applied. */
+export class HarnessPredicate<T extends ComponentHarness> {
+  private _predicates: AsyncPredicate<T>[] = [];
+  private _descriptions: string[] = [];
+
+  constructor(public harnessType: ComponentHarnessConstructor<T>) {}
+
+  /**
+   * Checks if a string matches the given pattern.
+   * @param s The string to check, or a Promise for the string to check.
+   * @param pattern The pattern the string is expected to match. If `pattern` is a string, `s` is
+   *   expected to match exactly. If `pattern` is a regex, a partial match is allowed.
+   * @return A Promise that resolves to whether the string matches the pattern.
+   */
+  static async stringMatches(s: string | Promise<string>, pattern: string | RegExp):
+      Promise<boolean> {
+    s = await s;
+    return typeof pattern === 'string' ? s === pattern : pattern.test(s);
+  }
+
+  /**
+   * Adds a predicate function to be run against candidate harnesses.
+   * @param description A description of this predicate that may be used in error messages.
+   * @param predicate An async predicate function.
+   * @return this (for method chaining).
+   */
+  add(description: string, predicate: AsyncPredicate<T>) {
+    this._descriptions.push(description);
+    this._predicates.push(predicate);
+    return this;
+  }
+
+  /**
+   * Adds a predicate function that depends on an option value to be run against candidate
+   * harnesses. If the option value is undefined, the predicate will be ignored.
+   * @param name The name of the option (may be used in error messages).
+   * @param option The option value.
+   * @param predicate The predicate function to run if the option value is not undefined.
+   * @return this (for method chaining).
+   */
+  addOption<O>(name: string, option: O | undefined, predicate: AsyncOptionPredicate<T, O>) {
+    const value = typeof option === 'string' ? `"${option}"` : `${option}`;
```

Comment:
Add a comment here like 
```ts
// Display quotation marks around string values.
```
It took me like 15 seconds to see how the ternary sub expressions were different